contracts/feirari/RewardsDistributorAdmin.sol

```diff
@@ -0,0 +1,135 @@
+
+import "../refs/CoreRef.sol";
+import "./IRewardsDistributorAdmin.sol";
+import "@openzeppelin/contracts/access/AccessControlEnumerable.sol";
+
+/// @notice this contract has its own internal RBAC. The reason for doing this
+/// and not leveraging core is twofold. One, it simplifies devops operations around adding
+/// and removing users, and two, by being self contained, it is more efficient as it does not need
+/// to make external calls to figure out who has a particular role.
+contract RewardsDistributorAdmin is IRewardsDistributorAdmin, CoreRef, AccessControlEnumerable {
+
+    bytes32 public constant override AUTO_REWARDS_DISTRIBUTOR_ROLE = keccak256("AUTO_REWARDS_DISTRIBUTOR_ROLE");
+
+    /// @notice rewards distributor contract
+    IRewardsDistributorAdmin public rewardsDistributorContract;
+
+    /// @param coreAddress address of core contract
+    /// @param _rewardsDistributorContract admin rewards distributor contract
+    /// @param _autoRewardDistributors list of auto rewards distributor contracts that can call this contract
+    constructor(
+        address coreAddress,
+        IRewardsDistributorAdmin _rewardsDistributorContract,
+        address[] memory _autoRewardDistributors
+    ) CoreRef(coreAddress) {
+        rewardsDistributorContract = _rewardsDistributorContract;
+        /// @notice Governance should create new admin role for this contract
+        /// and then grant this role to all AutoRewardsDistributors so that they can call this contract
+        /// The reason we are reusing the tribal chief admin role is it consolidates control, access
+        /// and means we don't have to create another OA timelock or multisig contract
+        _setContractAdminRole(keccak256("TRIBAL_CHIEF_ADMIN_ROLE"));
+        _setRoleAdmin(AUTO_REWARDS_DISTRIBUTOR_ROLE, DEFAULT_ADMIN_ROLE);
+
+        /// give all AutoRewardsDistributor contracts the proper role so that they can set borrow and supply speeds
+        for (uint256 i = 0; i < _autoRewardDistributors.length; i++) {
+            _setupRole(AUTO_REWARDS_DISTRIBUTOR_ROLE, _autoRewardDistributors[i]);
+        }
+    }
+
+    /**
+      * @notice Begins transfer of admin rights. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @dev Admin function to begin change of admin. The newPendingAdmin must call `_acceptAdmin` to finalize the transfer.
+      * @param newPendingAdmin New pending admin.
+      */
+    function _setPendingAdmin(address newPendingAdmin) external override onlyGovernor {
+        rewardsDistributorContract._setPendingAdmin(newPendingAdmin);
+    }
+
+    /**
+      * @notice Accepts transfer of admin rights. msg.sender must be pendingAdmin
+      * @dev Admin function for pending admin to accept role and update admin
+      */
+    function _acceptAdmin() external override {
+        rewardsDistributorContract._acceptAdmin();
+    }
+
+    /*** Comp Distribution ***/
+   /*** Comp Distribution Admin ***/
+
+    /**
```

Comment:
Add method that will allow guardian or governor to zero rewards for any cToken. This will be very useful in case an emergency shutdown is needed.